.eslintrc

```diff
@@ -6,19 +6,57 @@
   },
   "extends": [
     "semistandard",
-    "eslint:recommended",
-    "plugin:prettier/recommended"
+    "plugin:n/recommended", // Uses the recommended rules from eslint-plugin-n (node)
+    "plugin:@typescript-eslint/recommended", // Uses the recommended rules from the @typescript-eslint/eslint-plugin
+    "prettier" // enables eslint-plugin-prettier and displays prettier errors as ESLint errors. (must be the last config in extends array)
   ],
-  "ignorePatterns": ["node_modules/"],
+  "ignorePatterns": ["node_modules/", "artifacts", "dist"],
+  "parser": "@typescript-eslint/parser",
   "parserOptions": {
-    "ecmaVersion": "latest"
+    "project": "./tsconfig.json",
+    "ecmaVersion": 2018,
+    "sourceType": "module" // Allows for the use of imports
+  },
+  "settings": {
+    "import/parsers": {
+      "@typescript-eslint/parser": [".ts"]
+    },
+    "import/resolver": {
+      "typescript": {
+        "alwaysTryTypes": true, // always try to resolve types under `<root>@types` directory even it doesn't contain any source code, like `@types/unist`
```

Comment:
We use the `~/*` prefix in other frontend code instead of `@/*`, could we also do that here? Minor suggestion of course

Reply:
this alwaysTryTypes rule, is specifically for typescript types. it will just try to look for @types/ directory (if it exists) and will resolve types there that it cannot find anywhere else.

Reply:
Ah, it is for when using js files it seems. Nice to have, thanks for the explanation 👍